internal/pkg/cmd/authenticated_cli_command.go

```diff
@@ -176,27 +178,68 @@ func (c *AuthenticatedCLICommand) GetMetricsClient() (*ccloudv2.MetricsClient, e
 
 func (c *AuthenticatedCLICommand) GetSchemaRegistryClient() (*schemaregistry.Client, error) {
 	if c.schemaRegistryClient == nil {
-		ctx := c.Config.Context()
+		if c.Config.IsCloudLogin() {
+			clusters, err := c.V2Client.GetSchemaRegistryClustersByEnvironment(c.Context.GetCurrentEnvironment())
+			if err != nil {
+				return nil, err
+			}
+			if len(clusters) == 0 {
+				return nil, errors.NewSRNotEnabledError()
+			}
 
-		clusters, err := c.V2Client.GetSchemaRegistryClustersByEnvironment(ctx.GetCurrentEnvironment())
-		if err != nil {
-			return nil, err
-		}
-		if len(clusters) == 0 {
-			return nil, errors.NewSRNotEnabledError()
-		}
+			unsafeTrace, err := c.Flags().GetBool("unsafe-trace")
+			if err != nil {
+				return nil, err
+			}
 
-		dataplaneToken, err := auth.GetDataplaneToken(ctx.GetState(), ctx.GetPlatformServer())
-		if err != nil {
-			return nil, err
-		}
+			configuration := srsdk.NewConfiguration()
+			configuration.BasePath = clusters[0].Spec.GetHttpEndpoint()
+			configuration.DefaultHeader = map[string]string{"target-sr-cluster": clusters[0].GetId()}
+			configuration.UserAgent = c.Config.Version.UserAgent
+			configuration.Debug = unsafeTrace
+			configuration.HTTPClient = ccloudv2.NewRetryableHttpClient(unsafeTrace)
 
-		unsafeTrace, err := c.Flags().GetBool("unsafe-trace")
-		if err != nil {
-			return nil, err
-		}
+			dataplaneToken, err := auth.GetDataplaneToken(c.Context.GetState(), c.Context.GetPlatformServer())
+			if err != nil {
+				return nil, err
+			}
+
+			c.schemaRegistryClient = schemaregistry.NewClient(configuration, dataplaneToken)
+		} else {
+			schemaRegistryEndpoint, err := c.Flags().GetString("schema-registry-endpoint")
+			if err != nil {
+				return nil, err
+			}
+			if schemaRegistryEndpoint == "" {
+				return nil, errors.New(errors.SREndpointNotSpecifiedErrorMsg)
+			}
+
+			unsafeTrace, err := c.Flags().GetBool("unsafe-trace")
+			if err != nil {
+				return nil, err
+			}
 
-		c.schemaRegistryClient = schemaregistry.NewClient(clusters[0].Spec.GetHttpEndpoint(), c.Config.Version.UserAgent, unsafeTrace, dataplaneToken, clusters[0].GetId())
+			caLocation, err := c.Flags().GetString("ca-location")
+			if err != nil {
+				return nil, err
+			}
+			if caLocation == "" {
+				caLocation = auth.GetEnvWithFallback(auth.ConfluentPlatformCACertPath, auth.DeprecatedConfluentPlatformCACertPath)
+			}
+
+			client, err := utils.GetCAClient(caLocation)
+			if err != nil {
+				return nil, err
+			}
+
+			cfg := srsdk.NewConfiguration()
```

Comment:
```suggestion
			configuration := srsdk.NewConfiguration()
```

Nitpick: just to match the variable name in the other `if` branch.